### PR TITLE
Add `gulp-format`, use `depcheck`, and update deps

### PR DESCRIPTION
**Pull request by michplunkett**
## Describe your changes
I added the `gulp-prettier` package and added the `format` task to the end of the default `gulp` command. I also used the [`depcheck`](https://www.npmjs.com/package/depcheck) package to find any missing and unused dependencies. The base output is here and those are the ones I removed: 
```bash
michaelp@MacBook-Air-5 icj-project-rig % depcheck
Unused dependencies
* jquery
Unused devDependencies
* acorn
* caniuse-lite
* gulp-htmlmin
* gulp-if
* gulp-load-tasks
* request
Missing dependencies
* nav: ./src/scss/main.scss
michaelp@MacBook-Air-5 icj-project-rig % 
...
michaelp@MacBook-Air-5 icj-project-rig % depcheck
Unused dependencies
* jquery
michaelp@MacBook-Air-5 icj-project-rig % 
```

Ran `npm update ____` for these specific libraries:
- https://github.com/utdata/icj-project-rig/pull/86
- https://github.com/utdata/icj-project-rig/pull/85
- https://github.com/utdata/icj-project-rig/pull/84
- https://github.com/utdata/icj-project-rig/pull/83
- https://github.com/utdata/icj-project-rig/pull/81
- https://github.com/utdata/icj-project-rig/pull/77
- https://github.com/utdata/icj-project-rig/pull/76
- https://github.com/utdata/icj-project-rig/pull/75
- https://github.com/utdata/icj-project-rig/pull/69

## Checklist before requesting a review
- [x] All features within the repository still work.

```bash
michaelp@MacBook-Air-5 icj-project-rig % gulp       
[11:13:07] Using gulpfile ~/Documents/GitHub/icj-project-rig/gulpfile.js
[11:13:07] Starting 'default'...
[11:13:07] Starting 'clean'...
[11:13:07] Finished 'clean' after 667 μs
[11:13:07] Starting 'styles'...
[11:13:09] Finished 'styles' after 2.09 s
[11:13:09] Starting 'copy'...
[11:13:09] Finished 'copy' after 3.82 ms
[11:13:09] Starting 'lint'...
[11:13:09] Starting 'scripts'...
[11:13:09] Starting 'images'...
[11:13:09] Finished 'lint' after 339 ms
[11:13:09] gulp-imagemin: Minified 0 images
[11:13:09] Finished 'scripts' after 371 ms
[11:13:09] Finished 'images' after 372 ms
[11:13:09] Starting 'nunjucks'...
[11:13:09] Finished 'nunjucks' after 1.68 ms
[11:13:09] Starting 'bake'...
[11:13:09] Finished 'bake' after 1.85 ms
[11:13:09] Finished 'default' after 2.47 s
michaelp@MacBook-Air-5 icj-project-rig % gulp format
[11:13:14] Using gulpfile ~/Documents/GitHub/icj-project-rig/gulpfile.js
[11:13:14] Starting 'format'...
[11:13:14] Finished 'format' after 105 ms
michaelp@MacBook-Air-5 icj-project-rig % 
```

### CODE REVIEW

File: .prettierignore [hunk 1]
```diff
@@ -4,4 +4,6 @@ README.md
 .github
 
 # Directories and files to ignore
-docs/
+docs/css
+docs/img
+docs/js
```
Comment (michplunkett) [inline .prettierignore:9]: This way we format the html files within the docs directory.

File: package.json [hunk 2]
```diff
@@ -37,14 +37,14 @@
     "gulp-cache": "^1.1.3",
     "gulp-concat": "^2.6.1",
     "gulp-data": "^1.3.1",
-    "gulp-eslint": "^6.0.0",
```
Comment (michplunkett) [inline package.json:40]: Removed `gulp-eslint`.

File: package.json [hunk 3]
```diff
@@ -37,14 +37,14 @@
...
     "gulp-htmlmin": "^5.0.1",
     "gulp-if": "^3.0.0",
     "gulp-imagemin": "^8.0.0",
     "gulp-load-tasks": "^0.8.4",
     "gulp-newer": "^1.4.0",
     "gulp-nunjucks-render": "^2.2.3",
     "gulp-postcss": "^9.0.1",
+    "gulp-prettier": "^4.0.0",
```
Comment (michplunkett) [inline package.json:47]: Added `gulp-prettier` in its place.

File: tasks/format.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+import prettier from 'gulp-prettier';
+
+function format() {
+  return gulp
+    .src([
+      './src/js/*.js',
+      './docs/books/*.html',
+      './docs/index.html',
+      './src/js/*.js',
+      './src/data/*.json'
```
Comment (michplunkett) [inline tasks/format.js:11]: Only look at these file paths.

File: tasks/format.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+    ])
+    .pipe(prettier({ config: '.prettierrc.json' }))
```
Comment (michplunkett) [inline tasks/format.js:13]: Use the base config file.

File: tasks/format.js [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+    .pipe(gulp.dest((file) => file.base));
```
Comment (michplunkett) [inline tasks/format.js:14]: When changed, rewrite the file in-place.

File: package.json [hunk 7]
```diff
@@ -62,6 +56,7 @@
     "bootstrap": "^5.1.3",
     "fs-extra": "^11.1.1",
     "jquery": "^3.6.0",
```
Comment (michplunkett) [inline package.json:58]: `jquery` is pulled to the `docs/js` folder in a gulp step, so I left it here.

**Comment by michplunkett** [comment]
**Note:** There is an issue where `format` isn't changing the files since they do not exist yet (they should wait on the end of the promise), but the command on its own does work. I'm gonna take a look at it a little later when my brain 🧠 is fresh.

File: tasks/format.js [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+
+function format() {
+  return gulp
+    .src([
+      './src/js/*.js',
+      './src/data/*.json',
+      './docs/books/*.html',
+      './docs/*.html',
+      './src/js/*.js',
+      './src/data/*.json'
```
Comment (michplunkett) [inline tasks/format.js:12]: Wildcards are used in case files are not actually present in those directories.

File: tasks/serve.js [hunk 9]
```diff
@@ -28,7 +28,7 @@ function serve() {
   );
   gulp.watch(
     ['src/js/**/*.js'],
-    gulp.series('lint', 'scripts', browserSyncReload)
+    gulp.series('format', 'scripts', browserSyncReload)
```
Comment (michplunkett) [inline tasks/serve.js:31]: RIP `lint`.

File: .eslintrc.json [hunk 10]
```diff
@@ -1,6 +1,232 @@
+// Sourced from: https://chromium.googlesource.com/external/github.com/twbs/bootstrap/+/refs/tags/v4.1.2/.eslintrc.json
 {
-  "parserOptions": {
-    "ecmaVersion": 8,
-    "sourceType": "module"
+  "root": true,
+  "parser": "@babel/eslint-parser",
```
Comment (michplunkett) [inline .eslintrc.json:4]: `eslint-parser` is deprecated, so we're using `@babel/eslint-parser`.

File: tasks/lint.js [hunk 11]
```diff
@@ -1,12 +1,12 @@
+import browserSync from 'browser-sync';
+import eslint from 'gulp-eslint';
 import gulp from 'gulp';
 import gulpIf from 'gulp-if';
-import eslint from 'gulp-eslint';
-import browserSync from 'browser-sync';
 
 function lint() {
   return gulp
     .src(['src/js/**/*.js', '!node_modules/**'])
```
Comment (michplunkett) [inline tasks/lint.js:8]: I wasn't able to find a way that gets this working correctly, so I left it as it was.

File: README.md [hunk 12]
```diff
@@ -275,6 +275,7 @@ Gulp is the task runner and is configured in `gulpfile.js`. Individual tasks liv
 - `clear.js`: Clears out the gulp cache. Useful to reprocess images of the same name stuck in cache. Run `gulp clear` then re-run `gulp`.
 - `copy.js`: Used to copy production-necessary JavaScript files from `node_modules` into `docs/js`.
 - `fetch.js`: Downloads Google Drive files as data as configured in `project.config.json`.
+- `format.js`: Formats various `json`, `html`, and `js` files within the `src` and `data` folders.
```
Comment (michplunkett) [inline README.md:278]: Add `format.js` section to the README.

File: package.json [hunk 13]
```diff
@@ -53,15 +54,13 @@
     "imagemin-pngquant": "^6.0.1",
     "is-valid-glob": "^1.0.0",
     "journalize": "^2.5.1",
-    "request": "^2.88.2",
+    "prettier": "^2.8.8",
     "sass": "^1.49.8"
   },
   "dependencies": {
     "@popperjs/core": "^2.11.2",
-    "ansi-colors": "^4.1.3",
```
Comment (michplunkett) [inline package.json:61]: Moved these packages to `dev-dependencies`.

File: tasks/lint.js (hunk 11 above)
Comment (critmcdonald) [reply tasks/lint.js:8]: What was going on here that you wanted to change?

**Review by critmcdonald — COMMENTED** [review]
The unbalanced `<small>` tag is in `src/njk/_includes/footer.njk`. Sorry about that.

Again, not sure if you need anything else from me here. You have a conversation further into the review that I wasn't sure what you were trying to do (something with the lint.js file).

I'll just comment for now since I'm not sure about that one. Let me know and I'll come back and approve. I don't do a lot of pull request reviews so I'm just fumbling around.

File: tasks/lint.js (hunk 11 above)
Comment (michplunkett) [reply tasks/lint.js]: Whenever I ran `gulp lint` it never addressed any of the issues in the files. It also seems to require a file path style than the formatting task. Within the formatting task, I'd use `./src/js/**/*.js` and in here it's `src/js/**/*.js`.

**Comment by michplunkett** [comment]
Ran `gulp` and `gulp format` and it seems we're good to go! The output for both have been added to the PR description.

**Comment by michplunkett** [comment]
Once this is approved, I'd like to close the PRs that are listed in the description. I'll make one more PR where I run a `npm update` and then that SHOULD be it for updating packages.

File: src/njk/_includes/footer.njk [hunk 14]
```diff
@@ -1,6 +1,6 @@
 <footer>
 <p id="legal" class="center-block text-center">
-  <small>&copy; <span class="copyright-year"></span>
+  <small>&copy;</small> <span class="copyright-year"></span>
```
Comment (michplunkett) [inline src/njk/_includes/footer.njk:3]: The missing tag!

File: package.json (hunk 13 above)
Comment (jefflinwood) [reply package.json:63]: That definitely needed to be done.

File: src/njk/_includes/footer.njk (hunk 14 above)
Comment (jefflinwood) [reply src/njk/_includes/footer.njk:3]: Good catch!

**Review by jefflinwood — APPROVED** [review]
Approved!

**Pull request merged**